### PR TITLE
[#19037] feat: new design for manage nickname

### LINKED ISSUE #19037: Feature: New Design for Add Nickname for Contact
block by:
- #18963

User story:

5. **Adding a Nickname**
   - As a user, I want to add a nickname to the contacts in my network, so that I can remember or identify them more easily in my conversations.

33. **Assigning a Nickname to a Contact Requester**
    - As a user, I want to have the option to add a nickname to the contact requester's profile, so that I can personalize how I reference them before accepting the request.

51. **Assigning a Nickname to Sent Contact Request**
    - As Alisher, I want the ability to assign a nickname to Alicia's profile even while my contact request is pending, so that I can personalize my view of her profile.

functionality already implemented in current design

Scope:
- add nickname user flow

Figma: https://www.figma.com/file/jGs1P4RU4R1NRrY0xXmHC1/Profile-interactions?type=design&node-id=85-231496&mode=design&t=mAMotBHUP3dCSOyk-4

Image

### PR DESCRIPTION
**Pull request by mohsen-ghafouri**
fixes #19037

### Summary

- Add nickname user flow
- Remove nickname flow

### Result

https://github.com/status-im/status-mobile/assets/71308738/d41a5083-b604-4ad6-89b3-1bcbe4353965

status: ready

### CODE REVIEW

File: src/status_im/common/validation/profile.cljs [hunk 1]
```diff
@@ -50,3 +50,12 @@
     (has-special-characters? s) (i18n/label :t/are-not-allowed
                                             {:check (i18n/label :t/special-characters)})
     (bio-too-long? s)           (i18n/label :t/bio-is-too-long)))
+
+(defn validation-nickname
+  [s]
+  (cond
+    (or (= s nil) (= s ""))     nil
```
Comment (ibrkhalil) [inline src/status_im/common/validation/profile.cljs]: (or (= s nil) (= s ""))
To
(string/blank? s)

File: src/status_im/contexts/profile/contact/add_nickname/view.cljs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            [status-im.contexts.profile.utils :as profile.utils]
+            [utils.debounce :as debounce]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn view
+  []
+  (let [{:keys [public-key primary-name nickname customization-color]
+         :as   profile}                         (rf/sub [:contacts/current-contact])
+        ;; TODO: remove :blue when #18733 merged.
```
Comment (ibrkhalil) [inline src/status_im/contexts/profile/contact/add_nickname/view.cljs]: Can you please replace this todo with a GitHub issue link?

File: src/status_im/contexts/profile/contact/add_nickname/view.cljs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        :default-value  unsaved-nickname
+        :error?         (not (string/blank? error-msg))
+        :label          (i18n/label :t/nickname)
+        :on-change-text on-nickname-change}
+       :on-submit-editing on-nickname-submit]
+      [info-message/info-message
+       {:icon :i/info
+        :size :default
+        :type (if (not (string/blank? error-msg)) :error :default)}
+       (if (not (string/blank? error-msg))
```
Comment (ibrkhalil) [inline src/status_im/contexts/profile/contact/add_nickname/view.cljs]: Can be (if-not ...)
Or
(if (not-empty error-msg))

**Review by ibrkhalil — APPROVED** [review]

File: src/status_im/contexts/profile/contact/actions/view.cljs [hunk 4]
```diff
@@ -1,31 +1,58 @@
...
+          :label               (i18n/label :t/remove-nickname)
+          :on-press            on-remove-nickname
+          :add-divider?        true
+          :accessibility-label :remove-nickname
+          :danger?             true})
+       {:icon                :i/untrustworthy
+        :label               (i18n/label :t/mark-untrustworthy)
+        :on-press            not-implemented/alert
+        :accessibility-label :mark-untrustworthy
+        :add-divider?        (when (not has-nickname?) true)
```
Comment (seanstrom) [inline src/status_im/contexts/profile/contact/actions/view.cljs]: small nit 🔬 
maybe this could be `(when-not has-nickname? true)`?

**Review by seanstrom — APPROVED** [review]
Looks good! ✅ 
Left one tiny comment

**Pull request merged**